src/app/component/CartItem/CartItem.component.js

```diff
@@ -34,80 +35,98 @@ class CartItem extends Component {
         };
     }
 
+    componentWillUnmount() {
+        if (this.removeItem) this.removeItem.cancel();
+    }
+
     /**
      * Get link to product page
-     * @param url_key Url to product
-     * @return {{pathname: Srting, state Object}} Pathname and product state
+     * @return {{pathname: String, state Object}} Pathname and product state
      */
-    getProductLinkTo(url_key) {
-        if (!url_key) return undefined;
+    getProductLinkTo() {
         const {
-            product,
-            product: {
-                type_id,
-                configurable_options,
-                configurableVariantIndex,
-                parent,
-                variants = []
+            item: {
+                product,
+                product: {
+                    type_id,
+                    configurable_options,
+                    parent,
+                    variants = [],
+                    url_key
+                }
             }
         } = this.props;
 
         if (type_id === 'simple') return { pathname: `/product/${ url_key }` };
 
-        const { attributes } = variants[configurableVariantIndex];
+        const { product: { attributes } } = variants[this.getVariantIndex()];
 
-        const parameters = Object.entries(attributes).reduce(
-            (parameters, [code, { attribute_value }]) => {
-                if (Object.keys(configurable_options).includes(code)) return { ...parameters, [code]: attribute_value };
-                return parameters;
-            }, {}
-        );
+        const params = Array.prototype.reduce.call(configurable_options,
```

Comment:
It might be _undefined_. You must avoid writing fragile code. We are not in the full control of data, as well as this is base component, that will be re-used.